File: py/1245.py
from collections import defaultdict
from sys import stdin, stdout


def main() -> None:
    input = stdin.readline
    print = stdout.write
    while True:
        try:
            n = int(input())
        except ValueError:
            break
        pairs = defaultdict(lambda: defaultdict(int))
        count = 0
        for _ in range(n):
            m, l = input().split()
            pairs[m][l] += 1
        for size in pairs:
            aux = pairs[size]['D'] + pairs[size]['E']
            aux -= abs(pairs[size]['D'] - pairs[size]['E'])
            count += aux // 2
        print(str(count) + '\n')


if __name__ == '__main__':
    main()

File: cpp/1245.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

int main()
{
    int n, m, aux, count;
    char l;
    unordered_map<int, unordered_map<char, int>> pairs;
    while (cin >> n) {
        pairs.clear();
        count = 0;
        for (int i = 0; i < n; i++) {
            cin >> m >> l;
            pairs[m][l]++;
        }
        for (auto size: pairs) {
            aux = size.second['D'] + size.second['E'];
            aux -= abs(size.second['D'] - size.second['E']);
            count += aux / 2;
        }
        cout << count << '\n';
    }
    return 0;
}